src/contrib/cluster/Akka.Cluster.Tools/Client/ClusterClient.cs

```diff
@@ -440,6 +501,74 @@ private void PublishContactPoints()
 
             _contactPathsPublished = _contactPaths;
         }
+
+        private void ProcessSubscribeMessage(Subscribe subscribe, IActorRef receptionist)
+        {
+            if (RegisterExternalSubscriber(subscribe))
+                receptionist.Tell(new PublishSubscribe.Subscribe(subscribe.Topic, Self, subscribe.Group));
+        }
+
+        private void ProcessUnsubscribeMessage(Unsubscribe unsubscribe, IActorRef receptionist)
+        {
+            if (UnregisterExternalSubscriber(unsubscribe))
+                receptionist.Tell(new PublishSubscribe.Unsubscribe(unsubscribe.Subscribe.Topic, Self, unsubscribe.Subscribe.Group));
+        }
+
+        private bool RegisterExternalSubscriber(Subscribe subscribe)
+        {
+            ImmutableHashSet<IActorRef> actors;
+            if (!_externalSubscribers.TryGetValue(subscribe.MessageType, out actors))
+                actors = ImmutableHashSet<IActorRef>.Empty;
+
+            actors = actors.Add(subscribe.Subscriber);
+            _externalSubscribers = _externalSubscribers.SetItem(subscribe.MessageType, actors);
+
+            var key = new Tuple<string, string>(subscribe.Topic, subscribe.Group);
+            int subscribersCount;
+            _topicSubscriptions.TryGetValue(key, out subscribersCount);
+            _topicSubscriptions = _topicSubscriptions.SetItem(key, ++subscribersCount);
+
+            return subscribersCount == 1;
+        }
+
+        private bool UnregisterExternalSubscriber(Unsubscribe unsubscribe)
+        {
+            var subscribe = unsubscribe.Subscribe;
+            ImmutableHashSet<IActorRef> actors;
+            if (!_externalSubscribers.TryGetValue(subscribe.MessageType, out actors))
+                actors = ImmutableHashSet<IActorRef>.Empty;
+
+            if (!actors.Contains(subscribe.Subscriber))
+                return false;
+
+            actors = actors.Remove(subscribe.Subscriber);
+            if (actors.IsEmpty)
+                _externalSubscribers = _externalSubscribers.Remove(subscribe.MessageType);
+            else
+                _externalSubscribers = _externalSubscribers.SetItem(subscribe.MessageType, actors);
+
+            var key = new Tuple<string, string>(subscribe.Topic, subscribe.Group);
+            int subscribersCount;
+            _topicSubscriptions.TryGetValue(key, out subscribersCount);
+            if (--subscribersCount == 0)
+                _topicSubscriptions.Remove(key);
+            else
+                _topicSubscriptions = _topicSubscriptions.SetItem(key, subscribersCount);
+
+            return subscribersCount == 0;
+        }
+
+        private bool TryHandleExternalSubcriptions(object message)
+        {
+            var type = message.GetType();
+            ImmutableHashSet<IActorRef> actors;
+            if (_externalSubscribers.TryGetValue(type, out actors))
+            {
+                actors.ForEach(a => a.Tell(message));
```

Comment:
As mentioned before, we should contact with mediators instead of subscribers directly.